specs/kvstore/KeyValueStoreV2.fizz

```diff
@@ -0,0 +1,205 @@
+---
+deadlock_detection: false
+options:
+    max_actions: 200
+    max_concurrent_actions: 3
+    crash_on_yield: false
+action_options:
+    KeyValueStore.Put:
+        max_actions: 2
+    KeyValueStore.Get:
+        max_actions: 1
+    KeyValueStore.BackgroundFlushWAL:
+        max_concurrent_actions: 1
+    KeyValueStore.FreezeMemtable:
+        max_concurrent_actions: 1
+    KeyValueStore.FlushMemtableToL0:
+        max_concurrent_actions: 1
+---
+
+# Constants
+KEYS = ["k1"]
+VALUES = ["v0", "v1"]
+
+DURABILITY_MEMORY = 0
+DURABILITY_REMOTE = 2
+
+role KeyValueStore:
+    action Init:
+        self.wal_buffer = []
+        self.flushed_wal = []
+        self.memtable = []
+        self.immutable_memtable = []
+        self.l0 = []
```

Comment:
I see, `l0` and `flushed_wal` are made as the members of the KeyValueStore itself. This is generally not an issue as it abstracts away the ObjectStore/Bucket. But curious, if it is intentional. (If you define a separate role ObjectStoreBucket or Storage or something like that, it would become easier to visualize where the data is stored etc.